src/AcmHeaderPrototype/AcmHeaderPrototype.tsx

```diff
@@ -18,6 +18,35 @@ export type AcmHeaderPrototypeProps = {
     children: React.Component | React.ReactElement | React.ReactElement[]
 }
 
+function api<T>(url: string): Promise<T> {
+    return fetch(url).then((response) => {
+        if (!response.ok) {
+            throw new Error(response.statusText)
+        }
+        return response.json() as Promise<T>
+    })
+}
+
+function DropdownName() {
+    const [name, setName] = useState<string>('')
+
+    useEffect(() => {
+        const dev = process.env.NODE_ENV !== 'production'
+        const serverForTest = dev ? 'https://localhost:3000' : ''
+        api<{ username: string }>(`${serverForTest}/multicloud/common/username`)
+            .then(({ username }) => {
+                setName(username)
+            })
+            .catch((error) => {
+                // eslint-disable-next-line no-console
+                console.error(error)
+                setName('')
+            })
+    })
```

Comment:
I think you only want this to happen at the initial mounting of the component only?  If so, we can pass in a second argument to `useEffect` with `[]`:

https://reactjs.org/docs/hooks-effect.html

> Note
> 
> If you use this optimization, make sure the array includes all values from the component scope (such as props and state) that change over time and that are used by the effect. Otherwise, your code will reference stale values from previous renders. Learn more about how to deal with functions and what to do when the array changes too often.
> 
> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. This isn’t handled as a special case — it follows directly from how the dependencies array always works.
> 
> If you pass an empty array ([]), the props and state inside the effect will always have their initial values. While passing [] as the second argument is closer to the familiar componentDidMount and componentWillUnmount mental model, there are usually better solutions to avoid re-running effects too often. Also, don’t forget that React defers running useEffect until after the browser has painted, so doing extra work is less of a problem.
> 
> We recommend using the exhaustive-deps rule as part of our eslint-plugin-react-hooks package. It warns when dependencies are specified incorrectly and suggests a fix.
>

Reply:
@showeimer thanks for the tip! I updated the call accordingly